recipes/LibriSpeech/SSL/train.py

```diff
@@ -0,0 +1,518 @@
+                    if self.check_gradients(loss):
+                        self.scaler.step(self.optimizer)
+                    self.scaler.update()
+                    self.optimizer_step += 1
+                    self.optimizer.zero_grad()
+        else:
+            with self.no_sync(not should_step):
+                outputs = self.compute_forward(batch, Stage.TRAIN)
+                objectives = self.compute_objectives(
+                    outputs, batch, Stage.TRAIN
+                )
+                loss = objectives["loss"]
+                if self.hparams.diversity_loss_weight == 0.0:
+                    backprop_loss = loss
+                else:
+                    backprop_loss = (
+                        loss
+                        + objectives["diversity_loss"]
+                        * self.hparams.diversity_loss_weight
+                        * objectives["num_masked"]
+                    )
+                (backprop_loss / self.grad_accumulation_factor).backward()
+                objectives["total_loss"] = backprop_loss.detach()
+                if should_step:
+                    self.optimizer.step()
+                    self.optimizer.zero_grad()
+                    self.optimizer_step += 1
+
+        if should_step:
+            self.on_fit_batch_end(objectives)
+
+        return objectives["loss"].detach()
+
+    def on_fit_batch_end(self, objectives):
+        """ Called after fit_batch(), updates learning rate and does logging. """
+        if isinstance(self.modules.target_quantiser, DistributedDataParallel):
+            w2v_model = self.modules.target_quantiser.module
+        else:
+            w2v_model = self.modules.target_quantiser
+        w2v_model.quantiser.update_temp(self.optimizer_step)
+
+        self.hparams.lr_scheduler(self.optimizer, self.optimizer_step)
+
+        if (
+            hasattr(self.hparams, "log_interval")
+            and self.optimizer_step % self.hparams.log_interval == 0
+        ):
+
+            log_dct = {
+                k: (v.item() if isinstance(v, torch.Tensor) else v)
+                for k, v in objectives.items()
+            }
+            current_lr = self.optimizer.param_groups[0]["lr"]
+            log_dct["lr"] = current_lr
+            log_dct["avg_loss"] = self.avg_train_loss
+
+            if hasattr(self, "time_last_log"):
+                run_time_since_last_log = time.time() - self.time_last_log
+                log_dct["stats/run_time"] = run_time_since_last_log
+            self.time_last_log = time.time()
+
+            log_str = (
+                f"Optimizer step: {self.optimizer_step} - Objectives: {log_dct}"
+            )
+            logger.info(log_str)
+
+            if self.hparams.use_wandb:
+                run_on_main(
+                    wandb.log,
+                    kwargs={"data": log_dct, "step": self.optimizer_step,},
+                )
+
+    def evaluate_batch(self, batch, stage):
```

Comment:
Because accuracy is more meaningful and seems to correspond better to actual performance.
![w2v_loss_acc](https://user-images.githubusercontent.com/12561072/172213436-c58d3390-711c-4ca2-b219-4524dc509a83.png)

In the image above it is the model with the better accuracy (and worse loss) that achieves better WER after finetuning.